dev/benchmarks/microbenchmarks/lib/geometry/matrix_utils_transform_rect_bench.dart

```diff
@@ -0,0 +1,42 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math';
+
+import 'package:flutter/painting.dart';
+import 'package:flutter/rendering.dart';
+
+import '../common.dart';
+
+const int _kNumIters = 10000;
+
+void main() {
+  assert(false, "Don't run benchmarks in checked mode! Use 'flutter run --release'.");
+  print('MatrixUtils.transformRect benchmark...');
+
+  final Matrix4 transform = Matrix4.zero()
+    ..scale(1.2, 1.3, 1.0)
+    ..rotateZ(0.1);
```

Comment:
This produces a degenerate matrix which consists of all 0s and results in NaN for all coordinates.  The .zero() call should probably be replaced by .identity().